apps/api/src/app/preferences/preferences.controller.ts

```diff
@@ -57,7 +57,7 @@ export class PreferencesController {
         environmentId: user.environmentId,
         organizationId: user.organizationId,
         userId: user._id,
-        preferences: data.preferences,
+        preferences: buildWorkflowChannelPreferences(data.preferences),
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Is this an okay place to put this?

The issue is that this flows into a usecase in `application-generic` (which doesn't have `shared` as a dependency), so we can't use the centralized utility for building preferences after this point...‏